### PR TITLE
✨Add Tests for ManualTasks

### PR DESCRIPTION
**Pull request by alexanderkasten**
**Changes:**

1. Update dependencies
2. Add ManualTask Diagrams for tests
3. Add Tests for ManualTasks

PR: #56 

---

See the [examples](#example) for inspiration.

## How can others test the changes?

> Describe how others can test your changes (you can remove this section for typo fixes etc.)

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.).
- [x] I've rebased the `develop` branch with my branch before finishing this PR.
- [x] I've **summarized all changes** in a list above.
- [x] I've mentioned all **PRs, which relate to this one**.
- [x] I've prefixed my Pull Request title is according to [gitmoji guide](https://gitmoji.carloscuesta.me/).

## Example

<details>
<summary>
:warning: Before merging please click here to expand and follow the guide.
</summary>
<br>

Please use `:twisted_rightwards_arrows:` at the beginning of your merge commit title.

Example 1:

<pre>
<code>
:twisted_rightwards_arrows: :bug: Fix Wrong Text Decoration at ...
</code>
</pre>

To get your commit message, just copy the first part of this pull request.

Example 2:
<pre>
<code>
**Changes:**

- Fixes Wrong Text Decoration at ...
- Fixes some typos
- ...

**Issues:**

Closes #NumberOfFixedIssue

PR: #NumberOfThisPR
</code>
</pre>
</details>

## Emojis

<details>
<summary>
Expand for a list of most used Emojis.
</summary>
<br>

Please prefix your commit messages with an Emoji.

Ref: https://gitmoji.carloscuesta.me/

| Description              | Glyphe               | Emoji  |
|--------------------------|----------------------|--------|
| Bugfix                   | `:bug:`              | 🐛     |
| Fixing Security Issues   | `:lock:`             | 🔒     |
| Configuration releated   | `:wrench:`           | 🔧     |
| Cosmetic                 | `:lipstick:`         | 💄     |
| Dependencies Downgrade   | `:arrow_down:`       | ⬇️     |
| Dependencies Upgrade     | `:arrow_up:`         | ⬆️     |
| Formatting               | `:art:`              | 🎨     |
| Improving Performance    | `:zap:`              | ⚡️      |
| Initial commit           | `:tada:`             | 🎉     |
| Linter                   | `:rotating_light:`   | 🚨     |
| Miscellaneous            | `:package:`          | 📦     |
| New Feature              | `:sparkles:`         | ✨     |
| Refactoring Code         | `:recycle:`          | ♻️      |
| Releasing / Version tags | `:bookmark:`         | 🔖     |
| Removing Stuff           | `:fire:`             | 🔥     |
| Tests                    | `:white_check_mark:` | ✅     |
| Work In Progress (WIP)   | `:construction:`     | 🚧     |

</details>

### CODE REVIEW

File: _integration_tests/test/manual_tasks/get_correlation_manual_tasks.js [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+    should(manualTaskList.manualTasks.length).be.greaterThan(0);
+
+    const manualTask = manualTaskList.manualTasks[0];
+
+    manualTasksToFinishAfterTest.push(manualTask);
+
+    should(manualTask).have.property('id');
+    should(manualTask).have.property('correlationId');
+    should(manualTask).have.property('processModelId');
+    should(manualTask).have.property('tokenPayload');
```
Comment (ElRaptorus) [inline _integration_tests/test/manual_tasks/get_correlation_manual_tasks.js]: You could to the same assertions here, as you do in the test case above:

```js
    should(manualTask).have.property('id');
    should(manualTask).have.property('name');
    should(manualTask).have.property('correlationId');
    should(manualTask).have.property('processModelId');
    should(manualTask).have.property('processInstanceId');
    should(manualTask).have.property('tokenPayload');
```

These should all pass here as well.

File: _integration_tests/test/manual_tasks/finish_manual_tasks.js [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    const manualTaskList = await testFixtureProvider
+      .consumerApiClientService
+      .getManualTasksForProcessModelInCorrelation(defaultIdentity, processModelId, correlationId);
+
+    return manualTaskList.manualTasks[0];
+  }
+
+  it('should successfully finish the manual task.', async () => {
+
+    const manualTask = await createWaitingManualTask();
```
Comment (LeoTT) [inline _integration_tests/test/manual_tasks/finish_manual_tasks.js]: In the following(more specifically each `finishManualTask`-call) you keep using the properties of manualTask. Would destruct it here.

```ts
const {correlationId, flowNodeInstanceId, processInstanceId} = manualTask;
```

File: _integration_tests/test/manual_tasks/finish_manual_tasks.js [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+    console.log('manueltask from testfile', manualTask);
```
Comment (LeoTT) [inline _integration_tests/test/manual_tasks/finish_manual_tasks.js]: typo: manual

File: _integration_tests/test/manual_tasks/finish_manual_tasks.js [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+    
+    const manualTask = await createWaitingManualTask();
+    const invalidprocessInstanceId = 'invalidprocessInstanceId';
+
+    try {
+      await testFixtureProvider
+        .consumerApiClientService
+        .finishManualTask(defaultIdentity, invalidprocessInstanceId, manualTask.correlationId, manualTask.flowNodeInstanceId);
+
+      should.fail('unexpectedSuccesResult', undefined, 'This request should have failed!');
```
Comment (LeoTT) [inline _integration_tests/test/manual_tasks/finish_manual_tasks.js]: typo: success.
Goes for the following `should.fail` messages too.

File: _integration_tests/test/manual_tasks/get_correlation_manual_tasks.js [hunk 5]
```diff
@@ -0,0 +1,170 @@
+'use strict';
+
+const should = require('should');
+
+const TestFixtureProvider = require('../../dist/commonjs').TestFixtureProvider;
+const ProcessInstanceHandler = require('../../dist/commonjs').ProcessInstanceHandler;
```
Comment (LeoTT) [inline _integration_tests/test/manual_tasks/get_correlation_manual_tasks.js]: `const {TestFixtureProvider, ProcessInstanceHandler} = equire('../../dist/commonjs');` looks nicer(and also resembles es6-importsyntax more).

File: _integration_tests/test/manual_tasks/get_correlation_manual_tasks.js [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+
+  let processInstanceHandler;
+  let testFixtureProvider;
+
+  let defaultIdentity;
+
+  const processModelId = 'test_consumer_api_manualtask';
+  const processModelIdNoManualTasks = 'test_consumer_api_manualtask_empty';
+  const processModelIdCallActivity = 'test_consumer_api_manualtask_call_acvtivity';
+  const processModelIdCallActivitySubprocess = 'test_consumer_api_manualtask_call_acvtivity_subprocess';
```
Comment (LeoTT) [inline _integration_tests/test/manual_tasks/get_correlation_manual_tasks.js]: typos in both lines above: activity

**Pull request merged**